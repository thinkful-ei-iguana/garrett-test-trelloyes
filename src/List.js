import React from 'react';
import Card from './Card';
import './list.css';

function List(props) {
  return (
    <section className="List" key={props.listId}>
      <button type="button"
        className="List-add-button"
        onClick={() => props.onAddRandomCard(props.listId)}>
        + Add Random Card
      </button>
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">
        {props.cards.map(card =>
          <Card onDelete={props.onDelete}
            cardId={card.id}
            key={card.id}
            title={card.title}
            content={card.content} />)}
      </div>
    </section>
  )
}

export default List;