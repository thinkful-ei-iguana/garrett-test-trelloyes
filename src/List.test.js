import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('<List />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List key="a" header="myHeader" cards={[1, 2, 3]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    const tree = renderer.create(<List key="b" header="AmyHeader" cards={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});