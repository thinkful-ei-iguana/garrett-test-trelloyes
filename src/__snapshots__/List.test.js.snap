// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <button
    className="List-add-button"
    onClick={[Function]}
    type="button"
  >
    + Add Random Card
  </button>
  <header
    className="List-header"
  >
    <h2>
      AmyHeader
    </h2>
  </header>
  <div
    className="List-cards"
  />
</section>
`;
